tests/test_float_generator.py

```diff
@@ -0,0 +1,30 @@
+import unittest
+import numpy as np 
+from numpy.random import Generator, PCG64
+from synthetic_data.dataset_generators.float_generator import random_floats
+
+
+class TestRandomFloats(unittest.TestCase):
+    def setUp(self):
+        self.rng = Generator(PCG64(12345))
+
+    def test_return_type(self):
+        result = random_floats(self.rng)
+        self.assertIsInstance(result, np.ndarray)
+
+    def test_size(self):
+        num_rows = 5
+        result = random_floats(self.rng, num_rows = num_rows)
+        self.assertEqual(result.shape[0], num_rows)
+    
+    def test_values_range(self):
+        min_value, max_value = -1, 1
+        result = random_floats(self.rng, min_value, max_value)
+        self.assertTrue((result >= min_value).all() and (result <= max_value).all())
```

Comment:
readability here is a little tough, wonder if multiline for loop might be fine to make the test assert more clear. Not sure what is better, same with the assert for `test_sig_figs()`